friprosveta/views.py

```diff
@@ -478,6 +479,38 @@ def allocations_json(request, timetable_slug=None):
     filtered_allocations = _allocation_set(param_ids,
                                            tt.allocations,
                                            request.user.is_staff)
+
+    #returns more information about allocations
+    request_r = request.GET
+    if 'mode' in request_r:
+        mode_l = request_r['mode']
+        if mode_l == 'ext':
+            weekday_mapping = {wd[0]: i for i, wd in enumerate(WEEKDAYS)}
+            hour_mapping = {wh[0]: i for i, wh in enumerate(WORKHOURS)}
+            for allocation in filtered_allocations:
+                allocation.day_index = weekday_mapping[allocation.day]
+                allocation.hour_index = hour_mapping[allocation.start]
+            allocation_vms = sorted(filtered_allocations, key=lambda avm: avm.day_index)
+            allocations_by_day = [(d, list(avm_grouper))
+                                  for d, avm_grouper in itertools.groupby(allocation_vms, lambda avm: avm.day)]
+            allocations_ext = dict() 
+            for day, allocations in allocations_by_day:
+                allocations_day = []
+                for subject in allocations:
+                    teachers = [str(i) for i in subject.activityRealization.teachers.all()]
```

Comment:
The Allocation model already contains property teachers. Please use it like

`teachers = [str(i) for i in allocation.teachers.all()]`

.

Reply:
Code corrected.